segmentation/utils/spatial_analysis_utils.py

```diff
@@ -25,17 +26,54 @@ def calc_dist_matrix(label_map):
         centroids = np.array(a)
         dist_matrix = cdist(centroids, centroids)
         dist_mats_list.append(dist_matrix)
-    dist_mats = np.stack(dist_mats_list, axis=0)
+    # dist_mats = np.stack(dist_mats_list, axis=0)
     # label_map.coords["fovs"]
-    coords = [range(len(dist_mats)), range(dist_mats[0].data.shape[0]), range(dist_mats[0].data.shape[1])]
-    dims = ["points", "rows", "cols"]
-    dist_mats_xr = xr.DataArray(dist_mats, coords=coords, dims=dims)
-    return dist_mats_xr
+    # coords = [range(len(dist_mats)), range(dist_mats[0].data.shape[0]), range(dist_mats[0].data.shape[1])]
+    # dims = ["points", "rows", "cols"]
+    # dist_mats_xr = xr.DataArray(dist_mats, coords=coords, dims=dims)
 
+    with h5py.File("distance_mat.hdf5", "w") as f:
+        for i in range(0, len(dist_mats_list)):
+            key = 'dmat' + str(i)
+            dmat = f.create_dataset(key, data=dist_mats_list[i], dtype='uint16')
+    # dist_mats.close()
+    # return dist_mats
 
-def compute_close_cell_num(patient_data_markers, label_idx, thresh_vec,
-                           dist_mat, marker_num, dist_lim):
+
+def get_pos_cell_labels(analysis_type, pheno=None, fov_data=None,
```

Comment:
We need to add error checking here that all of the correct arguments are specified. For example, if analysis_type is cell_label, make sure that pheno is not None, etc etc.